### PR TITLE
Fixes #2322. Adds the feature to mark sources as favorite.

### LINKED ISSUE #2322: [Feature Request] Favorite Sources
**App version: 0.84**

**Android version: 9**

**Request:**
I have tons of sources (I basically download almost everyone) but I have a few that I like to see if there's something new.

Like the Manga Plus one... every now and then they release new mangas and every now and then I go there to check if I'm missing new mangas... same for other few sources I like.

I can either hide everything else or endlessly scroll to find those sources. If I'm searching for manga, then I would either have to un-hide the sources or do something else.

**Other details:**

By having favorite sources, I can check those whenever I want more easily and if I need to search new manga, those would show up first. 

Then after that could be the sources you already have manga saved.

After that all others.

It would help a lot when searching for manga or trying to see if your favorite scanlators started new projects.

Thank you all, love the app.

### PR DESCRIPTION
**Pull request by jatinAroraGit**
Added new database table Extensions which records the favorite extensions and all resolvers and type mappings related to it. Mark sources as favorite and the list of sources automatically gets sorted when a source is marked or unmarked as favorite. The favorite source is marked or unmarked by clicking the star shaped checkbox.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -234,7 +234,7 @@ dependencies {
 }
 
 buildscript {
-    ext.kotlin_version = '1.2.71'
+    ext.kotlin_version = '1.3.10'
```
Comment (hXtreme) [inline app/build.gradle]: Don't update the kotlin version here.

Comment (jatinAroraGit) [reply app/build.gradle]: I resolved it in the commit  https://github.com/inorichi/tachiyomi/pull/2350/commits/02045aa9532812b1adf765902be38c750f43ca92

Comment (jatinAroraGit) [reply app/build.gradle]: Resolved this in https://github.com/inorichi/tachiyomi/pull/2350/commits/ed2d8eedb25b8629519208fcb3375988a4886367

**Comment by joseph619** [comment]
@jatinAroraGit Can you please post an image or a GIF as an example to show how it works?

**Comment by Efreak** [comment]
Does this allow moving the 'Local' source/catalog to the top of the list?

**Comment by Soitora** [comment]
How's this going?

**Comment by joseph619** [comment]
@Efreak This is only for sources from extensions, Local manga is different

**Comment by arkon** [comment]
This should ideally be saved into preferences, not the database. The database should be things related to the library itself (i.e. roughly what gets backed up/restored).

**Pull request closed without merging**